var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -22,18 +23,61 @@ class Timemory(CMakePackage):
-    variant('tau', default=True, description='Enable TAU support')
-    variant('papi', default=True, description='Enable PAPI support')
-    variant('cuda', default=True, description='Enable CUDA support')
-    variant('cupti', default=True, description='Enable CUPTI support')
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('tau', default=False, description='Enable TAU support')
+    variant('papi', default=linux, description='Enable PAPI support')
+    variant('ompt', default=True, description='Enable OpenMP tools support')
+    variant('cuda', default=linux, description='Enable CUDA support')
+    variant('cupti', default=linux, description='Enable CUPTI support')
+    variant('tools', default=True, description='Build/install command line tools')
+    variant('vtune', default=False, description='Enable VTune support')
     variant('upcxx', default=False, description='Enable UPC++ support')
-    variant('gotcha', default=True, description='Enable GOTCHA support')
-    variant('likwid', default=True, description='Enable LIKWID support')
+    variant('gotcha', default=linux, description='Enable GOTCHA support')
+    variant('likwid', default=linux, description='Enable LIKWID support')
     variant('caliper', default=True, description='Enable Caliper support')
+    variant('dyninst', default=linux,
+            description='Build dynamic instrumentation tools')
+    variant('examples', default=False, description='Build/install examples')
     variant('gperftools', default=True,
             description='Enable gperftools support')
+    variant('kokkos_tools', default=True,
+            description='Build generic kokkos-tools libraries, e.g. kp_timemory, kp_timemory_filter')
+    variant('kokkos_build_config', default=False,
+            description='Build pre-configured (i.e. dedicated) kokkos-tools libraries, e.g. kp_timemory_cpu_flops')
+    variant('build_caliper', default=True,
+            description='Build Caliper submodule instead of spack installation')
+    variant('build_gotcha', default=False,
+            description='Build GOTCHA submodule instead of spack installation')
+    variant('build_ompt', default=True,
+            description='Build OpenMP/OMPT submodule instead of spack installation')
+    variant('cuda_arch', default='auto', description='CUDA architecture name',
+            values=('auto', 'kepler', 'tesla', 'maxwell', 'pascal',
+                    'volta', 'turing'), multi=False)
+    variant('cpu_target', default='auto',
+            description='Build for specific cpu architecture (specify cpu-model)')
+    variant('use_arch', default=False,
+            description='Build all of timemory w/ cpu_target architecture flags (default: roofline toolkit only)')
+    variant('tls_model', default='global-dynamic', description='Thread-local static model',
+            values=('global-dynamic', 'local-dynamic', 'initial-exec', 'local-exec'),
+            multi=False)
```

Comment:
I would prefer that variants (and possible values) use underscores instead of dashes to prevent confusion. Boolean variants can be disabled with `-foo`, so variants with dashes in the name are misleading.

Reply:
I understand that. The problem is that those values actually get translated to this directly from the CMake option to a C++ compiler flag: `add_cxx_flag_if_avail("-ftls-model=${TIMEMORY_TLS_MODEL}")` so it would necessitate doing something like `tls = tls.replace('_', '-')` which is the sort of generic string substitution I try to avoid whenever possible. I'll change it if you feel strongly but unless it has the potential to confuse the parser, I'd prefer to keep it as is.

Reply:
Oh, it won't confuse the parser, it just might confuse humans. It's your package, so you can do things how you want, but I don't think that string substitution will break unless you need options with both `-` and `_` in them.

Reply:
Yea I think anyone that actually wants to tweak this (advanced) option will likely be familiar with the options already